src/controllers/sessionLockController.ts

```diff
@@ -96,6 +99,48 @@ export default class SessionLockController {
               ),
               TE.chain(TE.fromEither)
             ),
+            // revoke pubkey
+            pipe(
+              TE.tryCatch(
+                () =>
+                  // retrieve the assertionRef for the user
+                  this.sessionStorage.getLollipopAssertionRefForUser(
+                    fiscalCode
+                  ),
+                E.toError
+              ),
+              TE.chain(TE.fromEither),
+              TE.chain(
+                flow(
+                  O.map(assertionRef =>
+                    TE.tryCatch(
+                      () =>
+                        // fire and forget the queue message
+                        new Promise<true>(resolve => {
+                          this.lollipopService
+                            .revokePreviousAssertionRef(assertionRef)
+                            .catch(constVoid);
+                          resolve(true);
+                        }),
+                      E.toError
+                    )
+                  ),
+                  // continue if there's no assertionRef on redis
+                  O.getOrElse(() => TE.of(true))
+                )
+              )
+            ),
+            // delete the assertionRef for the user
+            pipe(
+              TE.tryCatch(
+                () =>
+                  this.sessionStorage.delLollipopAssertionRefForUser(
+                    fiscalCode
+                  ),
+                E.toError
```

Comment:
Even if `getLollipopAssertionRefForUser` or `revokePreviousAssertionRef` returns left in the previous pipe we are calling `delLollipopAssertionRefForUser` here, are we ok with that?

Reply:
It is an applicative sequence. It should break if something goes left.